aries_vcx/src/protocols/typestate_con/invitee/mod.rs

```diff
@@ -0,0 +1,189 @@
+pub mod states;
+
+use std::sync::Arc;
+
+use messages::diddoc::aries::diddoc::AriesDidDoc;
+
+use crate::{errors::error::VcxResult, utils::uuid};
+
+use self::states::{initial::InitialState, invited::InvitedState, requested::RequestedState};
+
+use messages::{
+    a2a::A2AMessage,
+    concepts::ack::Ack,
+    protocols::connection::{request::Request, response::SignedResponse},
+};
+
+use super::{
+    common::states::{complete::CompleteState, responded::RespondedState},
+    initiation_type::Invitee,
+    pairwise_info::PairwiseInfo,
+    trait_bounds::Transport,
+    Connection,
+};
+use crate::{
+    common::signing::decode_signed_connection_response,
+    errors::error::{AriesVcxError, AriesVcxErrorKind},
+    handlers::util::verify_thread_id,
+    plugins::wallet::base_wallet::BaseWallet,
+};
+
+/// Convenience alias
+pub type InviteeConnection<S> = Connection<Invitee, S>;
+
+impl<S> InviteeConnection<S> {
+    pub fn new(source_id: String, pairwise_info: PairwiseInfo) -> InviteeConnection<InitialState> {
+        Connection {
+            source_id,
+            state: InitialState,
+            pairwise_info,
+            initiation_type: Invitee,
+        }
+    }
+}
+
+impl InviteeConnection<InitialState> {
+    // This should take an Invitation, but that also implies a DDO resolver
```

Comment:
good observation :-) Once we have a DDO Resolve trait https://github.com/hyperledger/aries-vcx/issues/706 I think that could improve the function signature here to. I wouldn't like state machines to depend on particular ddo resolving code, so passing some DdoResolver trait along with the invitation could perhaps be the ultimate solution

Reply:
That would indeed be the proper way to do it :-).